ui/app/components/app/ens-input.js

```diff
@@ -118,17 +124,36 @@ EnsInput.prototype.lookupEnsName = function (recipient) {
 EnsInput.prototype.componentDidUpdate = function (prevProps, prevState) {
   const state = this.state || {}
   const ensResolution = state.ensResolution
+  const networkHasEnsSupport = getNetworkEnsSupport(this.props.network)
   // If an address is sent without a nickname, meaning not from ENS or from
   // the user's own accounts, a default of a one-space string is used.
   const nickname = state.nickname || ' '
   if (prevProps.network !== this.props.network) {
-    const provider = global.ethereumProvider
-    this.ens = new ENS({ provider, network: this.props.network })
-    this.onChange(ensResolution)
+    if (networkHasEnsSupport) {
```

Comment:
With the removal of the `this.onChange(engResolution)` call, a few things change:
- `this.props.onChange()` is now only called once and only if `prevState && ensResolution && this.props.onChange &&
      ensResolution !== prevState.ensResolution` is true, previously it would be called in all cases that `this.onChange` was called
- previously, state would only be reset to falsy values (see line 143-151) and `this.checkName` only called if the recipient matched the `ensRE`. Now these happen on all networks that have ens support

A few other changes with regards to the order of calls. I have not traced down exactly what the implications are of this yet, but this suggests to me that we need to merge very carefully.

Perhaps we should write a list of expected behaviours for these components, and QA before and after these changes, and confirm that this introduces no new errors?